Ensure hdfs and yarn users are in appropriate hdfs and hadoop groups

cbaenziger opened the pull request:
This provides group membership for machines which already have hdfs and yarn users but not th necessary group membership for proper cluster operation.

**HDFS**: Needs to be in group hadoop (for proper updating of /var/log/); and should be in its self-named group for sanity's safe
**YARN**: Needs to be in group hadoop (for proper operation of the Linux Container Executor)

If one does not have this patch, one will see the datanode fail to start with errors in `/var/log/hadoop-hdfs/jsvc.err` like:
````
Initializing secure datanode resources
log4j:ERROR setFile(null,true) call failed.
java.io.FileNotFoundException: /var/log/hadoop-hdfs/hadoop-hdfs-datanode-hostname.log (Permission denied)
[...]
Opened info server at hostname/1.2.3.4:1006
Starting regular datanode initialization
Service exit with a return value of 1
````

amithkanand commented on cookbooks/bcpc-hadoop/recipes/datanode.rb:
```diff
@@ -33,6 +33,32 @@
+  # use manage as if the group does not exist
+  # we do not want an exception
+  action :manage
+  members ['hdfs', 'yarn']
```
Do we want to add `mapred` user here for `history server`?

cbaenziger replied:
I do not know why; I am unaware of any particular directory which is group `hadoop`?

cbaenziger replied:
We met and do not know of a case where `history_server` needs to be part of `mapred` so will leave it out.

amithkanand commented on cookbooks/bcpc-hadoop/recipes/datanode.rb:
```diff
@@ -102,6 +128,7 @@
   shell "/bin/bash"
   home "/usr/lib/hcatalog"
   supports :manage_home => false
+  not_if { user_exists? "hcat" }
```
This will not work as `user_exists` method is part of Kerberos pull request #117

amithkanand commented on cookbooks/bcpc-hadoop/recipes/hive.rb:
```diff
@@ -12,6 +12,7 @@
   shell "/bin/bash"
   home "/usr/lib/hcatalog"
   supports :manage_home => false
+  not_if { user_exists? "hcat" }
```
This will not work as user_exists method is part of Kerberos pull request #117

pu239ppy commented:
yarn needs to be added to mapred as well to it is able to write logs

org.apache.hadoop.yarn.server.nodemanager.LinuxContainerExecutor: Deleting path : /disk/0/yarn/logs/application_1439249650664_0001/container_1439249650664_0001_01_000001/stdout
2015-08-10 19:47:31,123 INFO org.apache.hadoop.yarn.server.nodemanager.LinuxContainerExecutor: Deleting path : /disk/0/yarn/logs/application_1439249650664_0001/container_1439249650664_0001_01_000001/syslog
2015-08-10 19:47:31,123 INFO org.apache.hadoop.yarn.server.nodemanager.LinuxContainerExecutor: Deleting path : /disk/0/yarn/logs/application_1439249650664_0001/container_1439249650664_0001_01_000001/stderr
2015-08-10 19:47:31,129 ERROR org.apache.hadoop.yarn.server.nodemanager.containermanager.logaggregation.AppLogAggregatorImpl: Failed to clean old logs
org.apache.hadoop.security.AccessControlException: Permission denied: user=yarn, access=EXECUTE, inode="/var/log/hadoop-yarn/apps/levtest":levtest:mapred:drwxrwx---
        at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.checkFsPermission(FSPermissionChecker.java:271)
        at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.check(FSPermissionChecker.java:257)
        at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.checkTraverse(FSPermissionChecker.java:208)
        at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.checkPermission(FSPermissionChecker.java:171)

cbaenziger commented:
Updates:
* Added `yarn` to the `mapred` group for proper log aggregation happiness
* Added a ulimit for Hive which we recently hit being too low and seemed appropriate here.

amithkanand commented:
LGTM
